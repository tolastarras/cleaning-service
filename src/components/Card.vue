<template lang="html">
  <v-container>
    <v-layout row wrap>
      <v-flex xs-12>
        <v-card flat>
          <v-img :src="require(`@/assets/card/${card.image}`)">
            <v-layout class="media-content"></v-layout>
          </v-img>
          <v-card-text>
            <h2 class="text-xs-center title" v-html="title"></h2>
            <div class="card-content">
              {{ card.content }}
            </div>
          </v-card-text>
          <v-card-actions class="justify-center">
            <v-btn dark outline color="blue" @click="readMore">Read More</v-btn>
          </v-card-actions>
        </v-card>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
export default {
  props: ['card', 'index', 'total'],
  computed: {
    title () {
      return this.$root.formatTitle(this.card.title)
    }
  },
  methods: {
    readMore () {
      this.$router.push('/about')
    }
  }
}
</script>

<style lang="scss" scoped>
.v-card {
  border: 1em solid white;
}

.v-image {
  height: 230px !important;
}

.card-content {
  margin: 1.2em 0;
  padding: 0 .4em;
  height: 5em !important;
  overflow: hidden;
  overflow-y: auto;
}

.v-card__text h2 {
  margin: .7em 0;
  width: 100% !important;
}

// .media-content {
//   background-color: rgba(0, 0, 0, 0.1);
// }

.v-card__actions {
  padding: 0 0 1em 0 !important;
}

.v-btn {
  margin: 0 auto;
  width: 100%;
}

// /* media queries */
// @media (max-width: 399px) {
//   .container {
//     padding-bottom: 0 !important;
//   }

//   .card__actions {
//     padding: 0 !important;
//   }

//   .card-content {
//     height: 6em !important;
//   }
// }

// @media only screen and (max-width: 599px) {
//   .card__text {
//     padding: 0 .4em !important;
//   }
// }

// @media (min-width: 960px) and (max-width: 1020px) {
//   .card__text > h2 {
//     letter-spacing: 8px !important;
//   }
// }

// @media (max-width: 1263px) {
//   .card-content {
//     height: 6em !important;
//   }
// }
</style>
