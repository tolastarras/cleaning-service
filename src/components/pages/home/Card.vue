<template lang="html">
  <v-container>
    <v-layout row wrap>
      <v-flex xs-12>
        <v-card flat>
          <v-card-media :src="card.src">
            <v-layout class="media-content"></v-layout>
          </v-card-media>
          <v-card-text>
            <h2 class="text-xs-center" v-html="title"></h2>
            <div class="card-content">
              {{ card.content }}
            </div>
          </v-card-text>
          <v-card-actions class="justify-center">
            <v-btn dark outline color="blue" @click="readMore">Read More</v-btn>
          </v-card-actions>
        </v-card>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
import { PROJECT_PATH } from '@/config/'
export default {
  props: ['card', 'index', 'total'],
  computed: {
    title () {
      return this.formatTitle(this.card.title)
    }
  },
  methods: {
    readMore () {
      this.$router.push(`${PROJECT_PATH}/about/`)
    }
  }
}
</script>

<style scoped lang="css">
h2 {
  text-transform: uppercase;
  font-size: 2em;
  padding: .4em 0 0 0;
  margin: 0;
}
b {
  letter-spacing: -5px;
  color: red;
}
.card {
  border: 1em solid white;
  z-index: 1;
}
.card__media {
  height: 230px !important;
}
.card-content {
  margin: 1.2em 0;
  padding: 0 .4em;
  height: 5em !important;
  overflow: hidden;
  overflow-y: auto;
}
.card__text h2 {
  margin-bottom: .5em;
  width: 100% !important;
}
.media-content {
  background-color: rgba(0, 0, 0, 0.1);
}
.card__actions {
  padding: 0 0 1em 0 !important;
}
.btn {
  margin: 0 auto;
  width: 100%;
}
h2 {
  color: green;
}
h2 > span {
  color: red;
}
/* media queries */
@media (max-width: 399px) {
  .container {
    padding-bottom: 0 !important;
  }
  .card__actions {
    padding: 0 !important;
  }
  .card-content {
    height: 6em !important;
  }
}
@media only screen and (max-width: 599px) {
  .card__text {
    padding: 0 .4em !important;
  }
}
@media (min-width: 960px) and (max-width: 1020px) {
  .card__text > h2 {
    letter-spacing: 8px !important;
  }
}
@media (max-width: 1263px) {
  .card-content {
    height: 6em !important;
  }
}
</style>
