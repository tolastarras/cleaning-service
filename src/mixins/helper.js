export default {
  methods: {
    formatTitle: str => {
      const [ left, right ] = str.split(' ')

      return `
      <div class="card-title text-uppercase font-weight-bold">
        ${left} <span>${right}</span>
      </div>`
    },
    getPage: path => {
      // remove double forward slashes from path and split url into array of words > 0
      let values = path
        // .replace(/\/\//g, '/')
        .split('/')
        .filter(param => param.length > 0)

      // return the last value (the page)
      return values[values.length - 1]
    }
  }
}
