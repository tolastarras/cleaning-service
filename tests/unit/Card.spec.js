import { shallowMount, createLocalVue } from '@vue/test-utils'

import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'

import Card from '@/components/Card'
import helper from '@/mixins/helper'

const localVue = createLocalVue()
localVue.use(Vuex)

Vue.use(Vuetify)

describe('Card', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Card, {
      localVue,
      mixins: [helper],
      propsData: {
        card: {
          title: 'Test',
          name: 'sink'
        }
      }
    })
  })

  it('is instantiated', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('renders without breaking', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
