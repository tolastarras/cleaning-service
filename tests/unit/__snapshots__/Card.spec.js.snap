// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders without breaking 1`] = `
<v-container-stub tag="div">
  <v-layout-stub tag="div" row="" wrap="">
    <v-flex-stub tag="div" xs-12="">
      <v-card-stub tag="div" flat="true">
        <v-img-stub src="" position="center center" transition="fade-transition">
          <v-layout-stub tag="div" class="media-content"></v-layout-stub>
        </v-img-stub>
        <v-card-text-stub>
          <h2 class="text-xs-center no-wrap">
            <div class="card-title text-uppercase font-weight-bold">
              Test <span>undefined</span>
            </div>
          </h2>
          <div class="card-content">

          </div>
        </v-card-text-stub>
        <v-card-actions-stub class="justify-center">
          <v-btn-stub color="blue" activeclass="v-btn--active" tag="button" dark="true" outline="true" type="button">
            Read More
          </v-btn-stub>
        </v-card-actions-stub>
      </v-card-stub>
    </v-flex-stub>
  </v-layout-stub>
</v-container-stub>
`;
